<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>前端监控</title>
  </head>
  <body>
    <div id="container">
      <div class="content">
        <input type="button" value="点击抛出错误" onclick="errorClick()" />
        <input type="button" value="点击抛出Promise错误" onclick="promiseErrorClick()" />
      </div>
    </div>
    <script>
      function errorClick() {
        window.someVar.error = 'error';
      }
      function promiseErrorClick() {
        new Promise(function (resolve, reject) {
          window.someVar.error = 'error';
          // reject();
        }).then(
          (res) => {
            console.log(res);
          }
          // (error) => {
          //   console.log('error'); // 写了失败回调函数就不会被捕获了
          // }
        );
      }
    </script>

    <script src="./someError.js"></script>
  </body>
</html>
